internal/services/recoveryservices/recovery_services_vault_data_source.go

```diff
@@ -4,78 +4,113 @@
+	Sku               string                                     `tfschema:"sku"`
+	Tags              map[string]string                          `tfschema:"tags"`
+}
 
-		Schema: map[string]*pluginsdk.Schema{
-			"name": {
-				Type:     pluginsdk.TypeString,
-				Required: true,
-			},
+func (SiteRecoveryRecoveryVaultDataSource) ModelObject() interface{} {
+	return &SiteRecoveryRecoveryVaultDataSourceModel{}
+}
+
+func (SiteRecoveryRecoveryVaultDataSource) ResourceType() string {
+	return "azurerm_recovery_services_vault"
+}
 
-			"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+func (SiteRecoveryRecoveryVaultDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.RecoveryServicesVaultName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+	}
+}
+func (SiteRecoveryRecoveryVaultDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationComputed(),
 
-			"location": commonschema.LocationComputed(),
+		"tags": commonschema.TagsDataSource(),
 
-			"tags": commonschema.TagsDataSource(),
+		"identity": commonschema.SystemAssignedUserAssignedIdentityComputed(),
 
-			"sku": {
-				Type:     pluginsdk.TypeString,
-				Computed: true,
-			},
+		"sku": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
 		},
 	}
 }
 
-func dataSourceRecoveryServicesVaultRead(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).RecoveryServices.VaultsClient
-	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
-	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
-	defer cancel()
+func (r SiteRecoveryRecoveryVaultDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.RecoveryServices.VaultsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
 
-	id := vaults.NewVaultID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
-	resp, err := client.Get(ctx, id)
-	if err != nil {
-		if response.WasNotFound(resp.HttpResponse) {
-			return fmt.Errorf("%s was not found", id)
-		}
+			var recoveryServiceVault SiteRecoveryRecoveryVaultDataSourceModel
+			if err := metadata.Decode(&recoveryServiceVault); err != nil {
+				return err
+			}
 
-		return fmt.Errorf("retrieving %s: %+v", id, err)
-	}
+			id := vaults.NewVaultID(subscriptionId, recoveryServiceVault.RecoveryGroupName, recoveryServiceVault.Name)
+			resp, err := client.Get(ctx, id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("recovery service vault %s not found", id.ID())
+				}
+				return fmt.Errorf("retreiving %s: %+v", id, err)
```

Comment:
```suggestion
				return fmt.Errorf("retrieving %s: %+v", id, err)
```

Reply:
What the reason to use `%+v` for err instead of `%w`?